lib/hsa/kernels/tracking_beamformer_nbeam.cl

```diff
@@ -3,7 +3,7 @@
 #define IM y
 #define NBEAM 1
```

Comment:
Just because I'd like to understand, do you understand what does this variable does?  It looks like `num_beams` is set by one of the `global_id`, so then what does `NBEAM` do?  My guess is it controls the number of beams done within the kernel, but honestly it's a bit confusing with `num_beams` being another variable going into this.

Reply:
Yeah I'm not sure why it's there either. It doesn't do anything:
```
for (int b=get_global_id(1)*NBEAM; b<(get_global_id(1)+1)*NBEAM; b++){
```
should I just remove it?

Reply:
Hmm, let's leave it for now.   I think it might be an optimization parameter.  I'll check with Keith about that kernel, since I think he was the last one to edit it.

Reply:
Okay